client/js/components/snippets/snippet.jsx

```diff
@@ -124,22 +130,25 @@ export default class Snippet extends React.Component {
       mime: this.props.language
     };
     let { style } = this.props;
-    let author = this.props.user.name;
-
+    let currentUser = this.getCurrentUser();
+    let author = this.props.user || currentUser;
+    let authorName = (author && author.name) || 'author';
```

Comment:
On create I think we should just mock `author` with `currentUser` and if user suddenly has lost session it's bigger problem than bad snippet create form displaying ;) we should definitely fix session problem in first place instead of creating such conditionals for safety. 
@blimusiek what do you think?